src/gc/collector.cpp

```diff
@@ -198,7 +207,7 @@ bool isValidGCObject(void* p) {
     return al->user_data == p && (al->kind_id == GCKind::CONSERVATIVE_PYTHON || al->kind_id == GCKind::PYTHON);
 }
 
-void setIsPythonObject(Box* b) {
+void registerPythonObject(Box* b) {
```

Comment:
I'm adding an assert in `DEFAULT_CLASS_SIMPLE` for class objects which you mention in #672